packages/nextjs/src/server/createGetAuth.ts

```diff
@@ -2,33 +2,94 @@ import type { AuthObject } from '@clerk/backend';
 import { constants } from '@clerk/backend/internal';
 import { isTruthy } from '@clerk/shared/underscore';
 
-import { withLogger } from '../utils/debugLogger';
+import { type LoggerNoCommit, withLogger } from '../utils/debugLogger';
+import { isNextWithUnstableServerActions } from '../utils/sdk-versions';
 import { getAuthDataFromRequest } from './data/getAuthDataFromRequest';
 import { getAuthAuthHeaderMissing } from './errors';
-import { getHeader } from './headers-utils';
+import { detectClerkMiddleware, getHeader } from './headers-utils';
 import type { RequestLike } from './types';
 import { assertAuthStatus } from './utils';
 
-export const createGetAuth = ({
-  noAuthStatusMessage,
+// Utility type to determine if a type is a Promise
+type IsPromise<T> = T extends Promise<any> ? true : false;
+
+const createGetAuth = <
```

Comment:
Good job building the conditional types used here.
It looks like the majority of the logic is with each `transformer` function. Would it be simpler to drop the transformer abstraction altogether, drop `createGetAuth`  and end up with 2 methods:
- `createSyncGetAuth`
- `createAsyncGetAuth`
(notice I moved the 'sync'/'async' adjective before the 'getAuth' part, which is a minor detail as the adjective describes the returned function and not the builder.

Duplicating a few lines would probably be worth it if we could remove an abstraction layer. What do you think?